src/connection/server_support.c

```diff
@@ -1289,6 +1289,46 @@ css_start_shutdown_server ()
   css_Server_shutdown_inited = true;
 }
 
+int
+css_register_ha_server (const char *server_name)
+{
+  assert (server_name != NULL);
+  assert (!HA_DISABLED ());
+
+  CSS_CONN_ENTRY *conn;
+  std::string message_to_master = css_pack_message_to_master (server_name);
```

Comment:
error handling. (NULL check)

Reply:
HA_MODE is not supported on the Windows platform. So, how about considering this case or just adding assert(false) temparally?

Reply:
return data type of `css_pack_message_to_master ()` is `std::string`.
but `css_pack_message_to_master ()` returns NULL for exception case, and there will be crash in that case. 
It should not return NULL using data type `std::string`. 

So, I think `css_pack_message_to_master ()` should be refactored first, and then we can handle the error case.

Reply:
Does it mean using "#if defined (WINDOWS)" to handle the case?

Reply:
Considering the Windows environment is already taken into account in the functions being called, 
it seems straightforward to add assert(false) using #if defined(WINDOWS). 

In the `hb_register_to_master()` function, an error will be returned if it is a Windows environment. When a `hb_register_to_master()` request is received in the cub_master thread in Windows, cub_master thread will send an error to the requester.

Reply:
I added `#if !defined (WINDOWS)` at the caller, similar to existing code in `css_init ()` where it calls `hb_register_to_master ()`.

Reply:
> return data type of css_pack_message_to_master () is std::string.
but css_pack_message_to_master () returns NULL for exception case, and there will be crash in that case.
It should not return NULL using data type std::string.

I think we can just make an issue to track it and go ahead.

Reply:
I added, http://jira.cubrid.org/browse/CBRD-24865